src/ParseServer.js

```diff
@@ -84,14 +87,20 @@ class ParseServer {
     databaseController
       .performInitialization()
       .then(() => hooksController.load())
-      .then(() => {
+      .then(async () => {
+        if (beforeSchemasMigration) {
+          await Promise.resolve(beforeSchemasMigration());
+        }
+        if (schemas) {
+          await new DefinedSchemas(schemas, this.config).execute();
+        }
```

Comment:
May be here we need to add an option to prevent auto migrate at startup and expose `new DefinedSchemas(schemas, this.config).execute();` on parse server instance to let a developer start migration by it self

Reply:
Migration is run by `serverStartComplete` user can add logic to decide when he want the migrations to be ran depending on his use case.